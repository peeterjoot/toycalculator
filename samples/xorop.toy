//THIS IS A GENERATED TEST CASE (./bitwiseop.perl --xor).  DO NOT EDIT

STRING s[16];
INT64 r;
BOOL i1;
INT16 i16;
INT32 i32;
INT64 i64;
INT8 i8;
BOOL j1;
INT16 j16;
INT32 j32;
INT64 j64;
INT8 j8;
i1 = TRUE;
i16 = 1000;
i32 = 100000;
i64 = 100000000000;
i8 = 10;
j1 = FALSE;
j16 = 100;
j32 = 10000;
j64 = 10000000000;
j8 = 1;
s = "i1 XOR j1";
PRINT s;
r = i1 XOR j1;
PRINT r;
s = "i1 XOR j16";
PRINT s;
r = i1 XOR j16;
PRINT r;
s = "i1 XOR j32";
PRINT s;
r = i1 XOR j32;
PRINT r;
s = "i1 XOR j64";
PRINT s;
r = i1 XOR j64;
PRINT r;
s = "i1 XOR j8";
PRINT s;
r = i1 XOR j8;
PRINT r;
s = "i16 XOR j1";
PRINT s;
r = i16 XOR j1;
PRINT r;
s = "i16 XOR j16";
PRINT s;
r = i16 XOR j16;
PRINT r;
s = "i16 XOR j32";
PRINT s;
r = i16 XOR j32;
PRINT r;
s = "i16 XOR j64";
PRINT s;
r = i16 XOR j64;
PRINT r;
s = "i16 XOR j8";
PRINT s;
r = i16 XOR j8;
PRINT r;
s = "i32 XOR j1";
PRINT s;
r = i32 XOR j1;
PRINT r;
s = "i32 XOR j16";
PRINT s;
r = i32 XOR j16;
PRINT r;
s = "i32 XOR j32";
PRINT s;
r = i32 XOR j32;
PRINT r;
s = "i32 XOR j64";
PRINT s;
r = i32 XOR j64;
PRINT r;
s = "i32 XOR j8";
PRINT s;
r = i32 XOR j8;
PRINT r;
s = "i64 XOR j1";
PRINT s;
r = i64 XOR j1;
PRINT r;
s = "i64 XOR j16";
PRINT s;
r = i64 XOR j16;
PRINT r;
s = "i64 XOR j32";
PRINT s;
r = i64 XOR j32;
PRINT r;
s = "i64 XOR j64";
PRINT s;
r = i64 XOR j64;
PRINT r;
s = "i64 XOR j8";
PRINT s;
r = i64 XOR j8;
PRINT r;
s = "i8 XOR j1";
PRINT s;
r = i8 XOR j1;
PRINT r;
s = "i8 XOR j16";
PRINT s;
r = i8 XOR j16;
PRINT r;
s = "i8 XOR j32";
PRINT s;
r = i8 XOR j32;
PRINT r;
s = "i8 XOR j64";
PRINT s;
r = i8 XOR j64;
PRINT r;
s = "i8 XOR j8";
PRINT s;
r = i8 XOR j8;
PRINT r;
