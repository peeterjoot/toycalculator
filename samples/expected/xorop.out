i1 XOR j1
1
i1 XOR j16
101
i1 XOR j32
10001
i1 XOR j64
10000000001
i1 XOR j8
0
i16 XOR j1
1000
i16 XOR j16
908
i16 XOR j32
9464
i16 XOR j64
10000001000
i16 XOR j8
1001
i32 XOR j1
100000
i32 XOR j16
100036
i32 XOR j32
106928
i32 XOR j64
9999901344
i32 XOR j8
100001
i64 XOR j1
100000000000
i64 XOR j16
100000000100
i64 XOR j32
99999993616
i64 XOR j64
90672270336
i64 XOR j8
100000000001
i8 XOR j1
10
i8 XOR j16
110
i8 XOR j32
10010
i8 XOR j64
10000000010
i8 XOR j8
11
